tests/src/integration/client.rs

```diff
@@ -0,0 +1,664 @@
+};
+use namada_apps::cli::utils::Cmd;
+use namada_apps::cli::{args, cmds, Context};
+use namada_apps::client::tx::submit_reveal_aux;
+use namada_apps::client::{rpc, tx};
+use namada_apps::wallet::cli_utils::{
+    address_add, address_key_add, address_key_find, address_list,
+    address_or_alias_find, key_and_address_gen, key_and_address_restore,
+    key_export, key_find, key_list, payment_address_gen,
+    payment_addresses_list, spending_key_gen, spending_keys_list,
+};
+
+use crate::e2e::setup::Bin;
+use crate::integration::node::MockNode;
+
+pub fn run(
+    node: &MockNode,
+    who: Bin,
+    mut args: Vec<&str>,
+) -> Result<(), Report> {
+    let cmd = match who {
+        Bin::Node => {
+            args.insert(0, "namadan");
+            let app = App::new("test");
+            let app = cmds::NamadaNode::add_sub(args::Global::def(app));
+            let matches = app.get_matches_from(args.clone());
+            cmds::Namada::Node(
+                cmds::NamadaNode::parse(&matches)
+                    .expect("Could not parse node command"),
+            )
+        }
+        Bin::Client => {
+            args.insert(0, "client");
+            let app = App::new("test");
+            let app = cmds::NamadaClient::add_sub(args::Global::def(app));
+            let matches = app.get_matches_from(args.clone());
+            cmds::Namada::Client(
+                cmds::NamadaClient::parse(&matches)
+                    .expect("Could not parse client command"),
+            )
+        }
+        Bin::Wallet => {
+            args.insert(0, "wallet");
+            let app = App::new("test");
+            let app = cmds::NamadaWallet::add_sub(args::Global::def(app));
+            let matches = app.get_matches_from(args.clone());
+            cmds::Namada::Wallet(
+                cmds::NamadaWallet::parse(&matches)
+                    .expect("Could not parse wallet command"),
+            )
+        }
+        Bin::Relayer => {
+            args.insert(0, "relayer");
+            let app = App::new("test");
+            let app = cmds::NamadaRelayer::add_sub(args::Global::def(app));
+            let matches = app.get_matches_from(args.clone());
+            cmds::Namada::Relayer(
+                cmds::NamadaRelayer::parse(&matches)
+                    .expect("Could not parse wallet command"),
+            )
+        }
+    };
+    let rt = tokio::runtime::Runtime::new().unwrap();
+
+    rt.block_on(node.handle_command(cmd))
+}
+
+impl MockNode {
+    async fn handle_command(
+        &self,
+        cmd: cli::cmds::Namada,
+    ) -> Result<(), Report> {
+        let global = {
+            let locked = self.shell.lock().unwrap();
+            Global {
+                chain_id: Some(locked.chain_id.clone()),
+                base_dir: locked.base_dir.clone(),
+                wasm_dir: Some(locked.wasm_dir.clone()),
+            }
+        };
+        let mut ctx = Context::new(global)?;
+        match cmd {
```

Comment:
for a future improvement it would be nice if we could re-use the existing definitions without repeating them here as we're not going through `apps/src/bin/namada-client/cli.rs` anymore with this